timemachine/cpp/src/bd_exchange_move.hpp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "device_buffer.hpp"
+#include "logsumexp.hpp"
+#include "nonbonded_mol_energy.hpp"
+#include "weighted_random_sampler.hpp"
+#include <array>
+#include <vector>
+
+namespace timemachine {
+
+// BDExchangeMove uses biased deletion to move waters randomly in a box. The reference implementation
+// is in timemachine/md/exchange/exchange_mover.py::BDExchangeMove
+template <typename RealType> class BDExchangeMove {
+
+private:
+    const int N_;
+    const int proposals_per_move_;
+    const int num_target_mols_;
+    const RealType beta_; // 1 / kT
+    NonbondedMolEnergyPotential<RealType> mol_potential_;
+    WeightedRandomSampler<RealType> sampler_;
+    LogSumExp<RealType> logsumexp_;
+    // Buffer for evaluating moves without touching the original coords
+    const DeviceBuffer<double> d_intermediate_coords_;
+    const DeviceBuffer<double> d_params_;
+    const DeviceBuffer<__int128> d_mol_energy_buffer_;
+    const DeviceBuffer<int> d_mol_offsets_;
+    const DeviceBuffer<RealType> d_log_weights_before_;
```

Comment:
why are these things const? Semantically seems weird (since the internals of them do change), i.e. if they were pointers `RealType * d_log_weights_before_`; they should not const-able since the internals change

Reply:
You are right, it is weird. Will remove the const

Reply:
Removed const in https://github.com/proteneer/timemachine/pull/1202/commits/07f449302f32806180e9540e0a7de8ba55d3dbff